app/models/user.rb

```diff
@@ -264,17 +264,20 @@ def invite_code=(code)
   end
 
   def password_required?
-    return false if Devise.pam_authentication || Devise.ldap_authentication
+    return false if external?
+
```

Comment:
This makes the tests fail. However, I actually do not understand the current tests. Where does “pam_user1” come from?

Reply:
(This specific test pass now that you updated it, but I'm still curious at how it works)

Reply:
Yeah it's not obvious but the fake PAM data is defined in config/environments/test.rb